### PR TITLE
Media Library: try grouping by date

### PR DESCRIPTION
**Pull request by iseulde**
**Note: see #10426 for an implementation with React Virtualized.**

---

This is a first run an adds structure only.

Unfortunately this change seems to require changes to `InfiniteList` as well, which I find quite messy. 😕 I haven’t found a cleaner way so far.

Another approach could be to group items in an element, but the % layout makes this difficult. It would still require changes to `InfiniteList`, but might make aligning easier.

To do:

- [x] Make sure date format is translated.
- [x] Show relative date in some cases (today, yesterday), add year for last year and older.
- [x] Add styling.
- [ ] Try to align small groups of images (maybe in separate branch).

See p5j4vm-1rz-p2.

### CODE REVIEW

File: client/my-sites/media-library/list.jsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ var ReactDom = require( 'react-dom' ),
 	findIndex = require( 'lodash/findIndex' );
 
 import { connect } from 'react-redux';
+import moment from 'moment';
```
Comment (youknowriad) [inline client/my-sites/media-library/list.jsx]: We could use our `localize` Higher Order Component which provides `moment` as a prop. Details here https://github.com/Automattic/i18n-calypso#localize

File: client/my-sites/media-library/list.jsx [hunk 2]
```diff
@@ -153,9 +154,8 @@ export const MediaLibraryList = React.createClass( {
 		return 'item-' + item.ID;
 	},
 
-	renderItem: function( item ) {
-		var index = findIndex( this.props.media, { ID: item.ID } ),
-			selectedItems = this.props.mediaLibrarySelectedItems,
+	renderItem: function( item, i, colIndex ) {
+		var selectedItems = this.props.mediaLibrarySelectedItems,
```
Comment (youknowriad) [inline client/my-sites/media-library/list.jsx]: I know it's legacy code, but It may be a good time to get rid of theses `var`s  and use `const` or `let`.

**Comment by iseulde** [comment]
Thanks for the feedback @youknowriad! I stumbled upon `VirtualList` while trying to upgrade `react-virtualized`, and I'll certainly see if the wrapper fits for this project.

File: client/my-sites/media-library/list.jsx (hunk 1 above)
Comment (iseulde) [inline client/my-sites/media-library/list.jsx]: Awesome, thanks for the tip. :)

**Comment by iseulde** [comment]
Here's an attempt with `react-virtualized`: 395908c25200f5f6cceafd7c35f83ac8bbc3c0bb. Note that this depends on #10265 and #10285, so I'll not create a separate PR before decisions are made for those.

**Comment by alisterscott** [comment]
Closing as this PR is stale - please reopen if necessary

**Pull request closed without merging**